pages/kiosk/stories/[story].tsx

```diff
@@ -0,0 +1,84 @@
+import React, { ReactNode } from 'react';
+import { useLanguage } from 'utils/useLanguageHook';
+import StoryReader from '../../../components/basecomponents/StoryReader';
+import stories from '../../../data/stories';
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
+import { GetStaticPaths, GetStaticProps } from 'next';
+import { ParsedUrlQuery } from 'querystring';
+import { Story } from '@types';
+import { getCountryVariant, Language } from '../../../utils/locales';
+import Custom404 from '../../404';
+import { StoryPhrase, getStoryData } from '../../../components/basecomponents/Story/storyStore';
+import { useSetAtom } from 'jotai';
+import { currentPlatformAtom } from 'components/basecomponents/Kiosk/atoms';
+import { Platform } from '@types';
+
+import withKioskLayout from 'utils/hoc/withKioskLayout';
+interface StoriesProps {
+  story: Story | undefined;
+  phrases: StoryPhrase[];
+}
+
+interface UrlParams extends ParsedUrlQuery {
+  story: string;
+}
+
+const StoriesContainer = ({ story, phrases }: StoriesProps): ReactNode => {
+  const { currentLanguage, otherLanguage } = useLanguage();
+  const setCurrentPlatform = useSetAtom(currentPlatformAtom);
+
+  if (!story) {
+    return 'Story not found';
+  }
+
+  if (!story.title[currentLanguage] || !story.title[otherLanguage]) {
+    return null;
+  }
+
+  const title_current = story.title[currentLanguage] || '';
+  const title_other = story.title[otherLanguage] || '';
+  setCurrentPlatform(Platform.KIOSK);
```

Comment:
This seems like an antipattern, no need to keep track of this manually, just check the URL to know if you are in kiosk or not.